bigtable/client/build_info.cc.in

```diff
@@ -0,0 +1,27 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "bigtable/client/build_info.h"
+
+namespace bigtable {
+inline namespace BIGTABLE_CLIENT_NS {
+
+char const compiler[] = R"""(@CMAKE_CXX_COMPILER_ID@ @CMAKE_CXX_COMPILER_VERSION@)""";
+
+char const compiler_flags[] = R"""(@CMAKE_CXX_FLAGS@ ${CMAKE_CXX_FLAGS_${BUILD_TYPE_UPPER}})""";
```

Comment:
This is going to get long, no? And it leaks local paths for, e.g., `-I` header paths, etc.? Are you sure you want to include that?

Reply:
I think this is Okay.   `CMAKE_CXX_FLAGS` is only `-Wall -Werror`, and `CMAKE_CXX_FLAGS_${BUILD_TYPE_UPPER}` is something like `-g` or `-O3 -ONDEBUG`.

The include paths and such do not go into these variables, you can put them there, but that is bad form, and our CMake files do not do that.

Reply:
SGTM, this works for me.